src/hover_provider/sourceHoverProvider.ts

```diff
@@ -123,6 +123,31 @@ export class SourceHoverProvider implements HoverProvider, Disposable {
       content.appendMarkdown(
         `<span style="color:#347890;">(source)&nbsp;</span><span><strong>${node.name}</strong></span>`,
       );
+      if (node.description !== "") {
+        content.appendMarkdown(`</br><span>${node.description}</span>`);
+      }
+      content.appendText("\n");
+      content.appendText("\n");
+      content.appendMarkdown("---");
+      content.appendText("\n");
+      content.appendText("\n");
+      for (const colKey in node.columns) {
+        const column = node.columns[colKey];
+        content.appendMarkdown(
+          `<span style="color:#347890;">(column)&nbsp;</span><span>${column.name} &nbsp;</span>`,
+        );
+        if (column.data_type !== null) {
+          content.appendMarkdown(
```

Comment:
I think we can consider refactoring the main function to be more readable and maintainable. Something in the lines of :

`private createTableContent(node: any): MarkdownString {
  const content = new MarkdownString();
  content.appendMarkdown(
    `<span style="color:#347890;">(source)&nbsp;</span><span><strong>${node.name}</strong></span>`,
  );
  if (node.description !== "") {
    content.appendMarkdown(`</br><span>${node.description}</span>`);
  }
  content.appendText("\n\n");
  content.appendMarkdown("---");
  content.appendText("\n\n");
  return content;
}

private appendColumnContent(content: MarkdownString, column: any) {
  content.appendMarkdown(
    `<span style="color:#347890;">(column)&nbsp;</span><span>${column.name} &nbsp;</span>`,
  );
  if (column.data_type !== null) {
    content.appendMarkdown(`<span>-&nbsp;${column.data_type.toUpperCase()}</span>`);
  }
  if (column.description !== "") {
    content.appendMarkdown(`<br/><span><em>${column.description}</em></span>`);
  }
  content.appendMarkdown("</br>");
}

private getSourceHover(
  sourceName: string,
  currentFilePath: Uri,
  tableName: string,
): MarkdownString | undefined {
  const projectRootpath = this.dbtProjectContainer.getProjectRootpath(currentFilePath);
  if (!projectRootpath) return;

  const sourceMap = this.sourceMetaMap.get(projectRootpath.fsPath);
  if (!sourceMap) return;

  const node = sourceMap.get(sourceName)?.tables.find(table => table.name === tableName);
  if (!node) return;

  const content = this.createTableContent(node);

  for (const colKey in node.columns) {
    const column = node.columns[colKey];
    this.appendColumnContent(content, column);
  }

  return content;
}
`

Note this is a sample code just to give you some idea of what I am thinking